tests/test_imod_coupler/test_ribametamod_cases.py

```diff
@@ -88,12 +87,8 @@ def set_confined_storage_formulation(
     return mf6_model
 
 
-def remove_sprinkling_from_groundwater(
-    msw_model: MetaSwapModel, mf6_model: Modflow6Simulation
-) -> MetaSwapModel:
+def remove_sprinkling_from_groundwater(msw_model: MetaSwapModel) -> MetaSwapModel:
     msw_model.pop("sprinkling")
-    msw_model.pop("mod2svat")
-    msw_model["mod2svat"] = msw.CouplerMapping(modflow_dis=mf6_model["GWF_1"]["dis"])
```

Comment:
Why did you remove this? Since the CouplerMapping was initialized with sprinkling activated, it includes those couplings. Thats why it is reinitialized.

Reply:
Because the CouplerMapping doesn't take any arguments anymore. The modflow packages are now requested upon writing, instead of upon initialization, in iMOD Python. In pseudo-code:

Before the refactor in iMOD Python:

```python
class CouplerMapping:
    def __init__(self, modflow_dis, modflow_wel=None):
         ...
    def write(self):
         ...    
```

After it:

```python
class CouplerMapping:
    def __init__(self):
         ...
    def write(self,modflow_dis, modflow_wel=None):
         ...    
```

Therefore the CouplerMapping object doesn't need to be re-initialized: mappings are created upon writing.